### DISCUSSION TITLE
FYI - Digital Ocean not very new customer friendly

### DISCUSSION THREAD

**Discussion by SenorKarlos** [onthegomap/planetiler#451] [General]
Tried to purchase a droplet as suggested in the planet.md file, and after sending them $5 to 'verify' me, they have denied me access. Any other providers that, you know, like money and aren't extremely expensive? I even used the referral link.

**Comment by wipfli** [comment]
AWS is like 2 dollars per hour. The many core machines require some increase of quata, but once you have it, it is very easy

**Reply by wipfli** [reply]
If you use it often, the hetzner server auction can also be nice

**Reply by ZeLonewolf** [reply]
$1 per hour.

**Reply by ZeLonewolf** [reply]
See example scripts at https://github.com/ZeLonewolf/planetiler-scripts

**Reply by msbarry** [reply]
That's unfortunate! I've never heard of that happening. I usually recommend digitalocean because there are no egress fees - with AWS it costs an extra $10 to move the result out. If you're using the result/serving traffic from AWS and don't need to actually get the mbtiles file afterwards then it's not a concern.

**Reply by SenorKarlos** [reply]
I just can't wrap my head around a business model that advertises all services then tells you, sorry, you need to be a customer for a while first. Like what I need to invent some other use just to qualify to do the business I showed up for? It's like those stupid employers that want college grad age but 10 years experience.

I'll take a look over those scripts and AWS I guess, I do need the results on my own server though so an extra $10 just to download sounds like more bullshit. Is SAAS and such always this convoluted???

**Reply by ZeLonewolf** [reply]
Do you need to pull down your mbtiles?  If not, you can just keep it within AWS and just run a tile server like I do:
https://6ug7hetxl9.execute-api.us-east-2.amazonaws.com/

You pay for EFS storage and a t3.nano or micro for the tile server and then just spin up the render server when you need updates.  That's been my approach. I think in total I pay something like $30 a month for the whole shebang.

**Reply by SenorKarlos** [reply]
I pay $0 for my home servers (well except power, but y'know). Just wanted to use Digital ocean for what they supposedly provide, but this seems to be too much for them. I've replied (and spammed a few more requests bc I'm annoyed lol), maybe they'll marten up.

**Reply by ZeLonewolf** [reply]
Sure, I do the same, but for public-facing servers, you kind of want that to be in a data center where nobody's accidentally stepping over the power cord.

**Reply by msbarry** [reply]
What's your home setup? Depending on your specs you might be able to run it on that?

You could also try linode, they've got similar machines and bandwidth pricing.

**Reply by SenorKarlos** [reply]
I'm currently running a Pokemon Go mapping service from home (and the power cords are very secure, thank you 😝😂 

The 'Engine' Server is: DELL R815 2U Server 4x AMD 6174 2.2Ghz 12-Core CPU H700 with 128GB DDR3 & 6 SATA drives in a RAID, and though it has tons of processing overhead, HDD bottleneck & in use

The 'Web' Server is: DELL R710 8 CPUs x Intel(R) Xeon(R) CPU E5540 @ 2.53GHz with 72GB DDR3 RAM, Raid over SAS drives, but currently making an OSM tile server, serving nominatim, and a few other tasks under an ESXi install that also hosts a hacintosh

Trying to do this at home is *possible*, but not very efficient when I can just throw money at a service. Digital Ocean can eat a.... sorry I digress.

I'm starting to think when I look at costs n hours etc... I should just grab a ded from TurnKey for a month 😂 before I bought servers they did well by me, and probably pretty comparable. I'll check out linode pricing and rules though too. AWS seems over my head a bit with them scripts lol

**Reply by SenorKarlos** [reply]
Linode looks promising, assuming they let me buy things beyond a toybox. I do have a couple output questions though.

The PLANET.md file notes a ~3h runtime and I notice the logs only mention up to z14. Can it do higher zoom levels, how, and what kind of time might that add?

The whole 'merge buildings at z13' - if I disable that for the time savings, it's not a huge hit to quality or anything eh?

**Reply by ZeLonewolf** [reply]
We turned it off in Americana because we felt we didn't need it, so feel free to look here:
https://zelonewolf.github.io/openstreetmap-americana

**Reply by wipfli** [reply]
I think Planetiler goes up to z15. I don't know if it can do more. Where I live, z14 is like 3 m precision https://docs.mapbox.com/help/glossary/zoom-level/

What is your use case for higher zoom levels / better precision?

**Reply by SenorKarlos** [reply]
My service produces static maps at z15+ and slippy map up to 18 (with a zoom hack to 20 in the actual map software lol)

**Reply by msbarry** [reply]
Planetiler can go to z15 but the openmaptiles profile currently only goes to 14. Those machines sound powerful enough to generate (12 cores, >64gb ram) but I'm not sure how it would run if you're not on ssd. 

The only quirk I remember from linode is you had to file a support ticket to use bigger machines. They answered it in a day or two though.

**Reply by ZeLonewolf** [reply]
The only real rationale for going deeper than z14 is if we think the size of the tiles is too problematic and we want to fix it by separating those features into sub-tiles.  Since we can run a z14 build in 40ish minutes, going to 15 or 16 is totally in the realm of possible.

**Reply by SenorKarlos** [reply]
how do I specify the zoom levels I want? I basically want to reproduce the z18 free download from 2017 available from maptiler (IIRC), but up to date and not for $150/mth lol.

**Reply by ZeLonewolf** [reply]
If you're using the openmaptiles profile, this is pointless -- you'd only be repeating data that's fully contained in the z14 tiles (and multiplying it by 256)

**Reply by SenorKarlos** [reply]
After digging around a bit I think I may have had a misconception about the downloads you can get or pay for from maptiler, comparing the currently available purchase size adds up to some of these world exports, and other projects denote z14 being enough. Thanks for the advice and making me research my assumptions! Assuming linode doesn't put up a hassle I think I'm good to try this lol

**Comment by SenorKarlos** [comment]
Can confirm Linode is better to deal with, they approved higher use with no hassles. Perhaps take a look into them and update the recommendations :D lol

I'm currently following https://github.com/onthegomap/planetiler/blob/main/PLANET.md#example and am using the standard dedicated 128GB RAM Linode, and it comes with 50 cores to use and 2500GiB storage, so more than ample. I'm thinking I could have gotten away with setting the java command to 110 or 120 for a little more performance, but it seems fine. 2 hours in and I'm at 25% of z13
